packages/plugin-ext/src/main/browser/editor/untitled-resource.ts

```diff
@@ -101,8 +101,8 @@ export class UntitledResource implements Resource {
         await this.fileResource.saveContents(content, options);
     }
 
-    async saveContentChanges(changes: TextDocumentContentChangeEvent[], options?: { encoding?: string, overwriteEncoding?: string }): Promise<void> {
-        if (!this.fileResource) {
+    async saveContentChanges(changes: TextDocumentContentChangeEvent[], options?: ResourceSaveOptions): Promise<void> {
+        if (!this.fileResource || !this.fileResource.saveContentChanges) {
```

Comment:
What do you mean by the second case?

The contract is that  if `saveContentChanges` is defined then it is supported if not then it is not. It is the same on master. I like an idea of capabilities, but it will be another refactoring.

Reply:
I guess that it depends on the underlying file system whether "saveContentChanges" is implemented. So the resource can't really decide that.